features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx

```diff
@@ -283,6 +284,38 @@ function EndpointOverview(props) {
         }
     }, [props]);
 
+    useEffect(() => {
+        if (endpointConfig && (endpointConfig.endpoint_type === 'http' || endpointConfig.endpoint_type === 'address')) {
+            if (
+                endpointCategory.prod
+                && endpointConfig.production_endpoints
+                && endpointConfig.production_endpoints.url && endpointConfig.production_endpoints.url !== ''
+                && endpointCategory.sandbox
+                && endpointConfig.sandbox_endpoints && endpointConfig.sandbox_endpoints.url !== ''
+            ) {
+                setIsEndpointUrlAvailable(true);
+            } else if (
+                endpointCategory.prod
+                && endpointConfig.production_endpoints
+                && endpointConfig.production_endpoints && endpointConfig.production_endpoints.url !== ''
+                && !endpointCategory.sandbox
+                && !endpointConfig.sandbox_endpoints
+            ) {
+                setIsEndpointUrlAvailable(true);
+            } else if (
+                endpointCategory.sandbox
+                && endpointConfig.sandbox_endpoints
+                && endpointConfig.sandbox_endpoints && endpointConfig.sandbox_endpoints.url !== ''
+                && !endpointCategory.production
+                && !endpointConfig.production_endpoints
+            ) {
+                setIsEndpointUrlAvailable(true);
+            } else {
+                setIsEndpointUrlAvailable(false);
+            }
```

Comment:
Shall we simplify this a bit: 
Example:
```suggestion

            var productionEpSpecified =  endpointCategory.prod
                        && endpointConfig.production_endpoints
                        && endpointConfig.production_endpoints.url && endpointConfig.production_endpoints.url !== '';
            var sandboxEpSpecified =  endpointCategory.sandbox
                && endpointConfig.sandbox_endpoints
                && endpointConfig.sandbox_endpoints.url && endpointConfig.sandbox_endpoints.url !== ''
            var productionEpNotSelected = !endpointCategory.sandbox && !endpointConfig.sandbox_endpoints;
            var sandboxEpNotSelected = !endpointCategory.sandbox && !endpointConfig.sandbox_endpoints;
            
            if ((productionEpSpecified && sandboxEpSpecified)
                 || (productionEpSpecified && sandboxEpNotSelected)
                 || (sandboxEpSpecified && productionEpNotSelected)) {
                setIsEndpointUrlAvailable(true);
            } else {
                setIsEndpointUrlAvailable(false);
            }
```

Will this work?

Reply:
fixed